test/spec_server.rb

```diff
@@ -141,7 +146,10 @@ def with_stderr
   it "check pid file presence and running process" do
     pidfile = Tempfile.open('pidfile') { |f| f.write($$); break f }.path
     server = Rack::Server.new(:pid => pidfile)
-    server.send(:pidfile_process_status).must_equal :running
+    server.send(:pidfile_process_status).must_equal :this_process
+
+    script = "print Rack::Server.new(:pid => \"#{pidfile}\").send(:pidfile_process_status).inspect"
+    `'#{ruby_exe}' -I./lib -rrack/server -e '#{script}'`.must_equal ":running"
```

Comment:
It was intended as an integration test to make sure it works end-to-end. Stubbing would turn this check into a less-powerful unit test. If someone wants to refactor this later and make improvements, it's fine.